modules/_labs/select/react/lib/Select.tsx

```diff
@@ -0,0 +1,342 @@
+  zIndex: 1,
+});
+
+const SelectWrapper = styled('div')<Pick<SelectProps, 'grow' | 'disabled'>>(
+  {
+    position: 'relative',
+  },
+  ({grow}) => ({
+    display: grow ? 'block' : 'inline-block',
+  }),
+  ({disabled}) => ({
+    '& span': {
+      backgroundColor: disabled ? colors.soap100 : colors.frenchVanilla100,
+    },
+    '&:hover path': {
+      fill: disabled ? undefined : colors.licorice500,
+    },
+  })
+);
+
+export default class Select extends React.Component<SelectProps, SelectState> {
+  private inputRef = React.createRef<HTMLInputElement>();
+
+  static ErrorType = ErrorType;
+
+  static defaultProps = {
+    disabled: false,
+  };
+
+  state: Readonly<SelectState> = {
+    focusedItemIndex: null,
+    justSelectedItemIndex: null,
+    label: '',
+    selectedItemIndex: 0,
+    showingMenu: false,
+  };
+
+  private indexByValue = (value: string): number => {
+    const childrenArray = React.Children.toArray(this.props.children);
+    return childrenArray.findIndex(child => child.props.value === value);
+  };
+
+  private toggleMenu = (show: boolean): void => {
+    if (show) {
+      this.setState({
+        focusedItemIndex: this.state.selectedItemIndex,
+        showingMenu: true,
+      });
+    } else {
+      this.setState({
+        focusedItemIndex: null,
+        showingMenu: false,
+      });
+    }
+  };
+
+  componentDidMount() {
+    const {children, value} = this.props;
+
+    const childrenArray = React.Children.toArray(children);
+
+    // if value exists, set state to that value...
+    if (value) {
+      const childIndex = this.indexByValue(value);
+
+      if (childIndex !== -1) {
+        this.setState({
+          label: childrenArray[childIndex].props.label,
+          selectedItemIndex: childIndex,
+        });
+        return;
+      }
+    }
+
+    // ... otherwise, set state to the first option
+    const firstOption = childrenArray[0];
+    // console.log('setting firstOption label:', firstOption.props.label);
+    this.setState({
+      label: firstOption.props.label,
+    });
+  }
+
+  handleSelectClick = (event: React.MouseEvent<HTMLInputElement>): void => {
+    this.toggleMenu(!this.state.showingMenu);
+  };
+
+  handleSelectBlur = (event: React.FocusEvent): void => {
+    this.toggleMenu(false);
+  };
+
+  handleOptionClick = (optionProps: SelectOptionProps): void => {
+    // disabled options cannot be clicked
+    if (optionProps.disabled) {
+      return;
+    }
+
+    const index = this.indexByValue(optionProps.value);
+    this.setState({
+      justSelectedItemIndex: index,
+      selectedItemIndex: index,
+    });
+
+    // offer visual feedback (flashing animation) briefly
+    // before dismissing menu; set new label when menu has been
+    // dismissed
+    setTimeout(() => {
+      this.setState({
+        justSelectedItemIndex: null,
+        label: optionProps.label,
+      });
+      this.toggleMenu(false);
+    }, dismissMenuTimeout);
```

Comment:
Anytime there is a `setTimeout`, `setInterval` or anything that has a callback that works outside the lifecycle of the component, it _must_ be removed when the component unmounts. Otherwise you'll get errors like "Cannot call `setState` on an unmounted component". React hooks make this easier by colocating setup and teardown code. In React component classes, setup and teardown are separated by lifecycle methods...